File: JS/MaximumNumbersOfBalloons.js
var maxNumberOfBalloons = function(text) {
    let freq = new Array(26).fill(0);

    for (let ch of text) {
        freq[ch.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }

    let mn = freq['b'.charCodeAt(0) - 'a'.charCodeAt(0)];
    mn = Math.min(mn, freq['a'.charCodeAt(0) - 'a'.charCodeAt(0)]);
    mn = Math.min(mn, freq['n'.charCodeAt(0) - 'a'.charCodeAt(0)]);
    mn = Math.min(mn, Math.floor(freq['l'.charCodeAt(0) - 'a'.charCodeAt(0)] / 2));
    mn = Math.min(mn, Math.floor(freq['o'.charCodeAt(0) - 'a'.charCodeAt(0)] / 2));

    return mn;
};

File: C/MaximumNumbersOfBalloons.c
int maxNumberOfBalloons(char* text) {
    int freq[26] = {0};

    for (int i = 0; text[i] != '\0'; i++) {
        freq[text[i] - 'a']++;
    }

    int min = freq['b' - 'a'];

    if (freq['a' - 'a'] < min) min = freq['a' - 'a'];
    if (freq['n' - 'a'] < min) min = freq['n' - 'a'];
    if (freq['l' - 'a'] / 2 < min) min = freq['l' - 'a'] / 2;
    if (freq['o' - 'a'] / 2 < min) min = freq['o' - 'a'] / 2;

    return min;
}